File: php/jsonproto/app/Benchmark/Dtos/TripDto.php
<?php

namespace App\Benchmark\Dtos;

use Carbon\CarbonImmutable;
use DateTimeImmutable;

class TripDto
{
    public function __construct(
        public DateTimeImmutable|CarbonImmutable $createdAt,
        public DateTimeImmutable|CarbonImmutable $updatedAt,
        public string $id,
        public string $code,
        public string $externalId,
        public string $department,
        public string $driverName,
        public string $driverMobile,
        public string $vehicleRegistration,
        public string $vehicleType,
        public DateTimeImmutable|CarbonImmutable $plannedStartAt,
        public DateTimeImmutable|CarbonImmutable $plannedEndAt,
        public DateTimeImmutable|CarbonImmutable $actualStartedAt,
        public DateTimeImmutable|CarbonImmutable $actualEndedAt,
        public TripStatusEnum $status,
        /** @var TransportDto $transports */
        public array $transports,
    )
    {}
}

File: java/src/main/java/io/belac/blog/dtos/TripDto.java
package io.belac.blog.dtos;

import io.quarkus.runtime.annotations.RegisterForReflection;

import java.time.OffsetDateTime;
import java.util.List;

@RegisterForReflection
public record TripDto (
    OffsetDateTime createdAt,
    OffsetDateTime updatedAt,
    String id,
    String code,
    String externalId,
    String department,

    String driverName,
    String driverMobile,
    String vehicleRegistration,
    String vehicleType,
    OffsetDateTime plannedStartAt,
    OffsetDateTime plannedEndAt,
    OffsetDateTime actualStartedAt,
    OffsetDateTime actualEndedAt,
    TripStatusEnum status,
    List<TransportDto> transports


    ){}
